test/OpenJ9_Jsr_292_API/src/com/ibm/j9/jsr292/api/MethodHandleAPI_defineClass.java

```diff
@@ -0,0 +1,88 @@
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+package com.ibm.j9.jsr292.api;
+
+import org.testng.annotations.Test;
+import org.testng.Assert;
+import java.lang.reflect.Method;
+import java.lang.invoke.MethodHandles.Lookup;
+import static java.lang.invoke.MethodHandles.lookup;
+import com.ibm.j9.jsr292.ClassBytes;
+
+public class MethodHandleAPI_defineClass {
+	final static Lookup lookup = lookup();
+	
+	/**
+	 * defineClass test for a null array of class types
+	 */
+	@Test(expectedExceptions = NullPointerException.class, groups = { "level.sanity" })
+	public static void test_defineClass_ClassBytes_NullArray() throws Throwable {
+		Class<?> targetClass = lookup.defineClass(null);
+		Assert.fail("The test case failed to detect a null array of class types");
+	}
+	
+	/**
+	 * defineClass test for a valid class byte array from a package different from the lookup class
+	 */
+	@Test(expectedExceptions = IllegalArgumentException.class, groups = { "level.sanity" })
+	public static void test_defineClass_ClassBytes_DifferentPackage() throws Throwable {
+		Class<?> targetClass = lookup.defineClass(ClassBytes.classBytes_DifferentPackage());
+		Assert.fail("The test case failed to detect that the class byte array belongs to a package different from the lookup class");
+	}
+	
+	/**
+	 * defineClass test for a lookup without the PACKAGE access mode
+	 */
+	@Test(expectedExceptions = IllegalAccessException.class, groups = { "level.sanity" })
+	public static void test_defineClass_Lookup_WithoutPackageMode() throws Throwable {
+		Lookup newLookup = lookup.dropLookupMode(Lookup.PACKAGE);
+		Class<?> targetClass = newLookup.defineClass(ClassBytes.validClassBytes_SamePackage());
+		Assert.fail("The test case failed to detect that the lookup doesn't have the PACKAGE access mode");
+	}
+	
+	/**
+	 * defineClass test for an corrupted class byte array
+	 */
+	@Test(expectedExceptions = LinkageError.class, groups = { "level.sanity" })
+	public static void test_defineClass_ClassBytes_CorruptedArray() throws Throwable {
+		Class<?> targetClass = lookup.defineClass(ClassBytes.invalidClassBytes_SamePackage());
+		Assert.fail("The test case failed to detect an corrupted class byte array");
+	}
+	
+	/**
+	 * defineClass test for a valid class byte array that shares the same class loader
+	 * , the same runtime package and the protection domain as the lookup class.
```

Comment:
Formatting nit: the comma is better after 'loader'.

Reply:
Updated.